sktime/forecasting/base/_base.py

```diff
@@ -2739,6 +2740,833 @@ def _predict(self, fh, X, y):
+        y : time series in ``sktime`` compatible format, optional (default=None)
+            Historical values of the time series that should be predicted.
+            If not None, global forecasting will be performed.
+            Only pass the historical values not the time points to be predicted.
+
+        Returns
+        -------
+        pred_dist : sktime BaseDistribution
+            predictive distribution
+            if marginal=True, will be marginal distribution by time point
+            if marginal=False and implemented by method, will be joint
+        """
+        # default behaviour is implemented if one of the following three is implemented
+        # TODO
+        # When we merge the _BaseGlobalForecaster back to BaseForecaster,
+        # we need to remove n=2
+        # _BaseGlobalForecaster is a child class of BaseForecaster,
+        # therefor, overwriting the method once is not indicating
+        # a concrete implementation which will cause infinite loop.
+        implements_interval = self._has_implementation_of("_predict_interval", n=2)
+        implements_quantiles = self._has_implementation_of("_predict_quantiles", n=2)
+        implements_var = self._has_implementation_of("_predict_var", n=2)
+        can_do_proba = implements_interval or implements_quantiles or implements_var
+
+        if not can_do_proba:
+            raise RuntimeError(
+                f"{self.__class__.__name__} does not implement "
+                "probabilistic forecasting, "
+                'but "capability:pred_int" flag has been set to True incorrectly. '
+                "This is likely a bug, please report, and/or set the flag to False."
+            )
+
+        # if any of the above are implemented, predict_var will have a default
+        #   we use predict_var to get scale, and predict to get location
+        pred_var = self._predict_var(fh=fh, X=X, y=y)
+        pred_std = np.sqrt(pred_var)
+        pred_mean = self.predict(fh=fh, X=X, y=y)
+        # ensure that pred_mean is a pd.DataFrame
+        df_types = ["pd.DataFrame", "pd-multiindex", "pd_multiindex_hier"]
+        pred_mean = convert_to(pred_mean, to_type=df_types)
+        # pred_mean and pred_var now have the same format
+
+        # default is normal with predict as mean and pred_var as variance
+        from sktime.proba.normal import Normal
+
+        index = pred_mean.index
+        columns = pred_mean.columns
+        pred_dist = Normal(mu=pred_mean, sigma=pred_std, index=index, columns=columns)
+
+        return pred_dist
+
+    @classmethod
+    def _has_implementation_of(cls, method, n=1):
+        """Check if method has a concrete implementation in this class.
+
+        This assumes that having an implementation is equivalent to
+            n overrides of `method` in the method resolution order.
+
+        Parameters
+        ----------
+        method : str, name of method to check implementation of
```

Comment:
please describe on new line, according to numpydoc, ie, starting from "name", should be new line

Reply:
done